ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -67,22 +28,16 @@ cpdef traceback_to_frames(traceback, max_nframes):
     :return: The serialized frames and the number of frames present in the original traceback.
     """
     tb = traceback
-    frames = []
-    nframes = 0
+
+    cdef list frames = []
+    cdef int nframes = 0
+    
     while tb is not None:
         if nframes < max_nframes:
```

Comment:
this is weird, should this be something more like:

```python
if nframes >= max_nframes:
    break
```

it seems the way this is written we will traverse the entire traceback every time, no matter what ?

Reply:
(I think Yun saw this behavior when they were debugging too)

Reply:
Yes I think the point is that we still want to count the number of frames, even though we then truncate the frame stack. Thats why we return `nframes` instead of just returning `frames` and deriving `nframes` from `len(frames)`.

Reply:
oh, you are right, we still want to return them all. ok I didn't quite notice that